src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -0,0 +1,207 @@
+package cromwell.engine.backend.jes
+
+import java.nio.file.{Path, Paths}
+
+import com.google.api.services.genomics.model.Parameter
+import com.typesafe.config.ConfigFactory
+import com.typesafe.scalalogging.LazyLogging
+import cromwell.binding.WdlExpression._
+import cromwell.binding._
+import cromwell.binding.types.WdlFileType
+import cromwell.binding.values._
+import cromwell.engine.backend.Backend
+import cromwell.engine.backend.Backend.RestartableWorkflow
+import cromwell.engine.db.DataAccess
+import cromwell.util.TryUtil
+import cromwell.util.google.GoogleCloudStoragePath
+import scala.concurrent.{Future, ExecutionContext}
+import scala.util.{Failure, Success, Try}
+import JesBackend._
+
+object JesBackend {
+  private lazy val JesConf = ConfigFactory.load.getConfig("backend").getConfig("jes")
+  lazy val GoogleProject = JesConf.getString("project")
+  lazy val GoogleApplicationName = JesConf.getString("applicationName")
+
+  lazy val JesConnection = JesInterface(GoogleApplicationName)
+
+  /*
+    FIXME: At least for now the only files that can be used are stdout/stderr. However this leads to a problem
+    where stdout.txt is input and output :) Redirect stdout/stderr to a different name, but it'll be localized back
+    in GCS as stdout/stderr. Yes, it's hacky.
+   */
+  val LocalStdoutParamName = "job_stdout"
+  val LocalStderrParamName = "job_stderr"
+
+  val LocalStdoutValue = "job.stdout.txt"
+  val LocalStderrValue = "job.stderr.txt"
+
+  // SOME MIGHT BE WRONG
+  val CromwellExecutionBucket = s"gs://cromwell-dev/cromwell-executions"
+
+  // Decoration around WorkflowDescriptor to generate bucket names and the like
+  implicit class JesWorkflowDescriptor(val descriptor: WorkflowDescriptor) extends AnyVal {
+    def bucket = s"$CromwellExecutionBucket/${descriptor.name}/${descriptor.id.toString}"
+    def callDir(call: Call) = s"$bucket/call-${call.name}"
+  }
+
+  // For now we want to always redirect stdout and stderr. This could be problematic if that's what the WDL calls stuff, but oh well
+  def standardParameters(callGcsPath: String): Seq[JesParameter] = Seq(
+    JesOutput(LocalStderrParamName, s"$callGcsPath/$LocalStderrValue", Paths.get(LocalStderrValue)),
+    JesOutput(LocalStdoutParamName, s"$callGcsPath/$LocalStdoutValue", Paths.get(LocalStdoutValue))
+  )
+
+  sealed trait JesParameter { // FIXME: Perhaps not the best name
+    def name: String
+    def gcs: String
+    def local: Path
+
+    final def isInput = this.isInstanceOf[JesInput]
+    final def isOutput = !isInput
+    // FIXME: Perhaps not the best name
```

Comment:
I swear the typos in here were things I fixed previously, c'est la vie.

I did want to have a conversation about some of the nomenclature here. 

The Trait is `JesParameter` and represents a name/gcs/local mapping. Except for the fact that Jes already has a `Parameter` class, I think this is an ok name?

However that's where the `toGoogleParameter` comes in, really it should be `toParameter` but then the Q is "wtf is a parameter?"

Looking at it now I think these are fine - does anyone object?